### DISCUSSION TITLE
ErrorBoundary info parameter

### DISCUSSION THREAD

**Discussion by dmitri-minkin** [remix-run/remix#5424] [Proposals]
## Description
It would be nice to have `info` parameter in the Remix ErrorBoundary in addition to `error` similar to [React ErrorBoundary](https://reactjs.org/docs/react-component.html#componentdidcatch) that has 
`componentDidCatch(error, info)`

## Why do we need this ?
We want to use Bugsnag to notify about errors. Bugsnag example for this where it uses both `error` and `info` parameters:

```ts
Bugsnag.notify(error, (event) => {
    event.addMetadata('react', info)
    event.severity = 'error'
    event.unhandled = true
})
```

## Combining Bugsnag and Remix ErrorBoundary
In general we would like to go even further with somehow combining Bugsnag and Remix ErrorBoundary in our app.

See discussion about custom ErrorBoundary and Bugsnag[here](https://github.com/bugsnag/bugsnag-js/issues/638)

Bugsnag has it's own ErrorBoundary defined [here](https://github.com/bugsnag/bugsnag-js/blob/next/packages/plugin-react/src/index.js#L66) which is a bit more powerful then just sending Bugsnag.notify(...)

**Comment by brophdawg11** [comment]
I think this makes sense to proxy along any additional metadata made available via the rendering layer.  `useRouteError` currently returns the error verbatim so I think there's 2 ways we could approach this to maintain backwards compatibility:

1. New hook that would expose an object with any additional meta into
   * `let { info } = useRouteErrorInfo()`
   * The object format allows us to expand beyond `info` down the road and keeps us flexible in any future alternate rendering library approaches.  For example, Vue gives you the [component instance and an `info` param](https://vuejs.org/api/composition-api-lifecycle.html#onerrorcaptured) so it would be nice to have a place to expose `instance` if/when needed.
2. New future flag that changes the API allowing us to expose info alongside the error.  
   * `future.v2_errorInfo = true`
   * `let { error, info } = useRouteError()`

I think I would lean towards 2 since it keeps the API surface a bit smaller
